python/lsst/ts/criopy/m1m3/acceleration_transformer.py

```diff
@@ -0,0 +1,107 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["AccelerationTransformer"]
+
+import pathlib
+from typing import Any
+
+import pandas as pd
+import yaml
+
+
+class AccelerationTransformer:
+    """
+    Reads M1M3 configuration. Transform raw values into calibrated values, and
+    X Y Z accelerations.
+
+    Paramaters
+    ==========
+    config_dir : `None | str | pathlib.Path = None`
+        Configuration directory.
+    """
+
+    class Accelerometer:
+        def __init__(self, config: Any):
+            self.bias = config["Bias"]
+            self.sensitivity = config["Sensitivity"]
+            self.offset = config["Offset"]
+            self.scalar = config["Scalar"]
+
+        def __matmul__(self, value: Any) -> Any:
+            return (value - self.bias) * self.sensitivity * self.scalar - self.offset
+
+    def __init__(self, config_dir: None | str | pathlib.Path = None):
+        self.accelerometeres: list[Any] = []
+        self.distances: list[float] = [0, 0, 0]
+
+        if config_dir is not None:
+            self.load_config(config_dir)
+
+    def load_config(self, config_dir: str | pathlib.Path) -> None:
+        """Load M1M3 configuration files.
+
+        Parameters
+        ----------
+        config_dir : `str | pathlib.Path`
+            Directory where to look for configuration files.
+        """
+        config_dir = pathlib.Path(config_dir)
+        config_file = config_dir / "_init.yaml"
+
+        with open(config_file, "r") as file:
+            config = yaml.safe_load(file)
+
+        ac = config["AccelerometerSettings"]
+
+        self.accelerometeres = []
+
+        for accelerometer in range(8):
+            self.accelerometeres.append(
+                self.Accelerometer(ac[f"Accelerometer{accelerometer + 1}"])
+            )
+
+        self.distances = [
+            ac["AngularAccelerationXDistance"],
+            ac["AngularAccelerationYDistance"],
+            ac["AngularAccelerationZDistance"],
+        ]
+
+    def calibrated(self, raw: pd.DataFrame) -> pd.DataFrame:
+        ret = pd.DataFrame()
+        print(raw)
+        for acc in range(8):
+            ret[f"accelerometer{acc}"] = (
+                self.accelerometeres[acc] @ raw[f"rawAccelerometer{acc}"]
+            )
+
+        return ret
+
+    def xyz(self, calibrated: pd.DataFrame) -> pd.DataFrame:
+        return pd.DataFrame(
+            {
+                "accelerationX": (calibrated.accelerometer7 - calibrated.accelerometer5)
+                / self.distances[0],
+                "accelerationY": (calibrated.accelerometer2 - calibrated.accelerometer0)
+                / self.distances[1],
+                "accelerationZ": (calibrated.accelerometer4 - calibrated.accelerometer0)
+                / self.distances[2],
```

Comment:
Please help to explain this part of calculation in code comment.

Reply:
Comment added